src/simperium/http-request.browser.js

```diff
@@ -10,7 +10,12 @@ export default function(
 		xhr.open( 'POST', url );
 		xhr.setRequestHeader( 'X-Simperium-API-Key', apiKey );
 
-		xhr.onload = () => resolve( xhr.responseText );
+		xhr.onload = () => {
+			const message = xhr.status === 429 && xhr.responseText === ''
+				? 'too many requests'
+				: xhr.responseText;
+			resolve(message);
+		};
```

Comment:
may sound silly given how simple this is right now, but I feel like a ladder of conditions would better set the stage for future changes here which might be likely. it's not the biggest issue, and there's a case to be made to let this be for now, but what are your thoughts about this construction instead?

```ts
xhr.onload = () => {
    if ( xhr.message === 429 && xhr.responseText = '' ) {
        return resolve('too many requests');
    }

    resolve(message);
}
```

you may find it silly for me to recommend changing a ternary into an `if` chain, but the early-abort here is key, that and the distinction that we're not really saying the message is one of two things, but rather that we're parsing the response and determining which of many possible outcomes we have.

Reply:
>the early-abort here is key, that and the distinction that we're not really saying the message is one of two things, but rather that we're parsing the response and determining which of many possible outcomes we have.

Your reasoning here makes total sense! It may not be 100% needed right now, but in the future, we may need to add more checks so I updated with your suggestion.